dGame/dInventory/Item.cpp

```diff
@@ -568,3 +570,43 @@ void Item::LoadConfigXml(const tinyxml2::XMLElement& i) {
 		config.push_back(LDFBaseData::DataFromString(value));
 	}
 }
+
+bool Item::CanDeleteItem(Item* item) {
+	if (!item) return false;
+	// TODO:
+	// Check if item is being possessed
+	// Check if the owner is mounting item? (how is this different than the above)
+	// Allow GM 9 to freely delete
+	// Finally, check Deletion Restriction
+	const auto& itemComponent = item->inventory->FindItemComponent(item->lot);
+	if (itemComponent.delResIndex == -1) return true;
+	return CheckDeletionRestriction(itemComponent.delResIndex, item->lot);
+}
+
+bool Item::CheckDeletionRestriction(uint32_t delResIndex, LOT item) {
+	auto* delresTable = CDClientManager::GetTable<CDDeletionRestrictionsTable>();
+	const auto restriction = delresTable->GetByID(delResIndex);
+
+	switch(restriction.checkType) {
+		case eDeletionRestrictionsCheckType::INCLUDE_LOTS:
+			if (std::ranges::find(restriction.ids, item) != restriction.ids.end()) return false;
+			else return true;
+		case eDeletionRestrictionsCheckType::EXCLUDE_LOTS:
+			if (std::ranges::find(restriction.ids, item) != restriction.ids.end()) return true;
+			else return false;
+		case eDeletionRestrictionsCheckType::ANY_OF_THESE:
+			// TODO: Implement
+			return true;
+		case eDeletionRestrictionsCheckType::ALL_OF_THESE:
+			// TODO: Implement
+			return true;
+		case eDeletionRestrictionsCheckType::WHILE_IN_ZONE:
+			if (std::ranges::find(restriction.ids, Game::zoneManager->GetZoneID().GetMapID()) != restriction.ids.end()) return false;
+			else return true;
+		case eDeletionRestrictionsCheckType::ALWAYS_RESTRICTED:
+			return false;
+		case eDeletionRestrictionsCheckType::MAX:
+		default:
+			return true;
+	}
```

Comment:
I would consider instead a bool toReturn before the switch and a single return at the bottom of the function instead.  This would also allow you to greatly simplify the logic in the loop

Reply:
no, that would make too much sense /s

Reply:
One of reasons for the verbose-ness of this currently is that I wanted to make sure I got the logic correct. now that I have, I will simplify it